Airside_Functions.cpp

```diff
@@ -0,0 +1,458 @@
+/**************************************************************************************************/
+// Author: Jingting Liu
+// April 9th, 2021
+
+// to run the test without cmake:
+// gcc -g Airside_Functions.cpp
+// a.out
+
+// to run the test with cmake:
+// mkdir build, cd build
+// cmake ..
+// make
+// execute generated target
+
+
+// refer to this page for the stucture of mavlink messages 
+// https://mavlink.io/en/guide/serialization.html
+
+// The decoder and encoder only support GPS and gimbal control, other simpler commands will be taken
+// care of by the Xbee communication directly
+/**************************************************************************************************/
+
+#include "Mavlink2_lib/common/mavlink.h"
+#include "Airside_Functions.hpp"
+
+mavlink_decoding_status_t Mavlink_airside_decoder(PIGO_Message_IDs_e* type, uint8_t incomingByte, uint8_t *telemetryData)
+{
+    int channel = MAVLINK_COMM_0; //mavlink default one channel
+    PIGO_Message_IDs_e decoded_message_type = MESSAGE_ID_NONE;
+    mavlink_decoding_status_t decoding_status = MAVLINK_DECODING_INCOMPLETE;
+
+    mavlink_status_t status;
+    memset(&status, 0x00, sizeof(mavlink_status_t));
+
+    mavlink_message_t decoded_msg;
+    memset(&decoded_msg, 0, sizeof(mavlink_message_t));
+
+    uint8_t message_received = mavlink_parse_char(channel, incomingByte, &decoded_msg, &status);
```

Comment:
Perhaps a quick comment saying that this method will parse the incoming byte and store it in decoded_msg will be helpful for future members to understand :)) 

Or a link to here would suffice to: http://docs.ros.org/en/indigo/api/mavlink/html/include__v2_80_2mavlink__helpers_8h.html#ad91e8323cefc65965574c09e72365d7d